### ISSUE TITLE
Accept header check doesn't support `*/*`

### ISSUE THREAD

**Issue by ddossot** [nidi3/raml-tester#39]
We've tried upgrading to RAML tester `0.8.2` but have many tests failing because the new `Accept` header test doesn't cater to `*/*`.

Example (only relevant parts are kept):
```json
{
    "requestHeaders": {
        "Accept": [
            "*/*"
        ]
    },
    "responseHeaders": {
        "Content-Type": [
            "application/json; charset=UTF-8"
        ]
    },
    "responseViolations": [
        "Response Content-Type 'application/json; charset=UTF-8' is not compatible with Accept header '*/*'"
    ]
}
```
The specification of the [`Accept`](http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.1) header allows complex expressions like: `Accept: audio/*; q=0.2, audio/basic`. Thus, IMO, if RAML Tester decides to enforce `Accept`/`Content-Type` validity, it has to fully support `Accept`'s spec (FWW I'm using the Java helper in [this library](https://code.google.com/p/mimeparse/) for that intent in some of my apps).

**Comment by nidi3** [comment]
Complex accept values are supported. The problem is matching the response content-type with the correct mim-type in the raml.
Working on it.

**Comment by ddossot** [comment]
OK. So maybe the problem is different? The RAML spec defines `application/json` as the response `Content-Type`, not `application/json; charset=UTF-8`. Maybe the issue comes from the fact `charset` is added?

If I look at the 50 other violations we got with `0.8.2` I see most of them are similar: the server responded with a `Content-Type` like `application/vnd.unbounce.api.v0.4+json; charset=utf-8` but the spec has `application/vnd.unbounce.api.v0.4+json`.

Of course, we don't want to specify the encoding in the specs, it's more an implementation detail than a real part of the contract so I think we need RAML Tester to give a little leeway here.

**Comment by nidi3** [comment]
That was exactly the problem.
The content-type of the response is now matched with the *nearest compatible* mime-type in the RAML.
So your examples should work now.

Is it possible for you to test your RAMLs against the current trunk? Would be nice to have more security before the next release...)

**Comment by ddossot** [comment]
I'm happy to confirm this is fixed in version `0.8.3-SNAPSHOT` :+1:
